### PR TITLE
Add mergify for additional checks

### PR DESCRIPTION
**Pull request by okurz**
After this PR is merged pull requests can be automatically merged with a special label and as soon as all rules specified are met. For example: Only if all checks pass, a verification run was provided, no review pending or asking for changes, at least two approvals given, no label to wait, etc. Additional checks are conducted and show up as check in github status details same as other checks. Pull requests can still be merged manually as is but this approach helps for a couple of reasons:
* Acceptance criteria for merge are codified into rules that everybody can read
* Nobody needs to wait anymore if everything was already provided and approved, verification runs have been checked but you just wait for the CI to finish
* No trivial things are overlooked in review which can be automatically checked
* The rules can be extended in the future to cover further checks. For example I plan to automatically check provided verification runs or use the feature to semi-automatically trigger verification runs and wait for their successful completion

You can try out how mergify behaves within the test pull request https://github.com/okurz/os-autoinst-distri-opensuse/pull/4

Related progress issues:
* https://progress.opensuse.org/issues/72841
* https://progress.opensuse.org/issues/49502
* https://progress.opensuse.org/issues/48641

### CODE REVIEW

**Comment by okurz** [comment]
@mergifyio rebase

**Comment by okurz** [comment]
@os-autoinst/tests-maintainer as this should simplify all your work I am asking you for review.

**Comment by coolgw** [comment]
LGTM

**Comment by jlausuch** [comment]
LGTM

File: .mergify.yml
Comment (b10n1k) [inline .mergify.yml:19]: i would suggest to add `work-in-progress` label here https://docs.mergify.com/examples/#id9

**Review by b10n1k — CHANGES_REQUESTED** [review]
i love this. i was asking for all of those 3y of mine in the company :). now reviews can be more careful from now on. 
My first thought however is how do you influence the flow in case you need more reviews for some reason. Thats why i suggest the use the label rule. 
Also i would like to ask you to append the documentation. I think `https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#preparing-a-new-pull-request` is the right place, but i am not sure. 
Also i have a question. How this works when you have you update the PR and you have already one aprroval? Do it cancel the last review and two new approvals require or it doesnt change anything?

**Comment by ggardet** [comment]
LGTM!

File: .mergify.yml
Comment (b10n1k) [inline .mergify.yml:6]: Sorry i just notice that label is already applied. But i think the WIP is missing.

Comment (okurz) [reply .mergify.yml:6]: Sorry I don't know what you mean

File: .mergify.yml
Comment (Vogtinator) [inline .mergify.yml:37]: I always found this text weird. The emoji doesn't really fit, or do I just not understand it?

File: .mergify.yml
Comment (Vogtinator) [inline .mergify.yml:2]: As observed in https://github.com/os-autoinst/os-autoinst/pull/1842#issuecomment-976578261, one major issue with mergify is that pushes to the pull request do not invalidate existing approvals. Thus it's possible to push broken (or even malicious) changes between review and merge.

**Review by Vogtinator — CHANGES_REQUESTED** [review]

File: .mergify.yml
Comment (pdostal) [reply .mergify.yml:6]: Some people still use `[WIP] ` as the prefix either in pull request title or even in commit message.

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by alice-suse** [comment]
LGTM

**Comment by lemon-suse** [comment]
I like this idea, it will save everyone's time. LGTM.

File: .mergify.yml
Comment (pevik) [reply .mergify.yml:6]: I'd ignore commit message and match only label.

**Review by foursixnine — CHANGES_REQUESTED** [review]
I'm really not a fan of enabling mergify for the tests, while it would be amazing and a lot of time saving, unless one can have a list of submitters whose's PR's can be automerged.

Reasoning for this: We run this on openSUSE and SLE as is, so if somebody misses checking a zipped file (of which we have many), or some random binary... we can end up in a bad place, as @Vogtinator mentions: 

People can change things between approvals and add malicious code...

FUD aside, the other side of the problem is while we only would need one or two persons to approve, someone else's code might break. You break it, you fix it, is cool though.

File: .mergify.yml
Comment (jknphy) [reply .mergify.yml:6]: some people do not have merge permissions to use labels, the check for WIP in the title still apply here, right?

**Comment by OleksandrOrlov** [comment]
Sorry, but I don't think this is a good approach of merging things automatically, even with 2 approvals. And I'm afraid it may bring more troubles then benefits to our codebase. I've seen a lot of times when a PR received a couple of "Approvals" but then some new reviewer came and raised a valid point to change the PR and a lot of people change their mind. 

Also, nothing prevents from altering a PR after receiving "Approve". How to deal with such cases?

One more thing is that it makes "Approve" button to be taken more serious then before. So, enabling auto-merge also introduces artificial limitations in review process, and I guess a lot of people will just avoid them by typing LGTM without pressing "Approve" button:)

**Comment by jknphy** [comment]
-1 I think that it could potentially break the work flow of some teams, in the sense that even users with no privileges to merge (which cannot even set labels) can approve it, right? so at the end it is easy that some PR, where WIP or label is missed can be approved by someone less experience and even approved by someone external to the team interested in the change but not really knowledgeable of the code, not giving the opportunity to more senior colleagues to take a look. It could be also a waste of time to have to sync with other colleagues about githhub features to adapt to some teams, for example requesting things like 'set the label here for me' or 'do this other github feature here' instead of sync about the works itself.

File: .mergify.yml
Comment (okurz) [reply .mergify.yml:2]: That's correct but IMHO has nothing to do with mergify. I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something. It's still in the responsibility of reviewers to only approve if the PR is *actually* acceptable

**Comment by b10n1k** [comment]
> I'm really not a fan of enabling mergify for the tests, while it would be amazing and a lot of time saving, unless one can have a list of submitters whose's PR's can be automerged.
> 
> Reasoning for this: We run this on openSUSE and SLE as is, so if somebody misses checking a zipped file (of which we have many), or some random binary... we can end up in a bad place, as @Vogtinator mentions:
> 
> People can change things between approvals and add malicious code...
>

IMO this is a consideration of the team's process. Do not approve early on without making sure that review and CI are finished. i doubt that many are checking the changes done after the approve. So even now a malicious code or whatever can pass through in theory.  
 
> FUD aside, the other side of the problem is while we only would need one or two persons to approve, someone else's code might break. You break it, you fix it, is cool though.
> 
> So, from my pov:
> 
> * Until we have an automated check that the pr is not adding compressed files
> * Until we have an automated check that there are no external things being loaded
> 
> It's a no from me.

File: .mergify.yml
Comment (okurz) [reply .mergify.yml:37]: The common meaning is "please", see https://emojipedia.org/folded-hands/ . I would prefer to keep the very same text we use for other repos in https://github.com/os-autoinst

File: .mergify.yml
Comment (okurz) [reply .mergify.yml:6]: yes, WIP is handled by https://github.com/apps/wip and any WIP in the PR subject or git commit message will still block merge

**Comment by okurz** [comment]
> -1 I think that it could potentially break the work flow of some teams, in the sense that even users with no privileges to merge (which cannot even set labels) can approve it, right?

No, this does not count as real "approval".

> ok I just verified that this only can happen when that person is asked to review by somone with privileges.

correct

> But at the end it is easy that some PR, where WIP or label is missed can be approved by someone less experience and even approved by someone external to the team interested in the change but not really knowledgeable of the code, not giving the opportunity to more senior colleagues to take a look.

Well, mistakes can happen if this is even a realistic risk. However I am sure people would learn quickly that approval is something to take serious. Everyone can write "+1" if they are unsure.

> It could be also a waste of time to have to sync with other colleagues about github features, for example requesting things like 'set the label here for me' or 'do this other github feature here' or clarify why is a 'request for change' and not a suggestion, instead of syncing about the works itself.

I am not sure what you mean here. There is no need to discuss about github features if you don't want to use them. All the old workflows can still work. Only if really all rules are met then mergify would merge automatically. The author can prevent that with a mere "WIP" in the subject, in the git commit message, asking for review, etc., any of these will work

File: .mergify.yml
Comment (Vogtinator) [reply .mergify.yml:2]: > I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something.

IMO that option makes sense and should be enabled. Approvals on earlier revisions are still visible, so nothing is lost by doing that, but there is a distinction between "this state was reviewed" and "someone reviewed some older, possibly totally different code in the past", which is necessary for a proper decision by humans as well as bots.

> It's still in the responsibility of reviewers to only approve if the PR is actually acceptable

If the PR is "actually acceptable", reviewers can also hit the "Merge after pipeline succeeds" button, which is pretty much equivalent to adding an approval and waiting for mergify, just that there is no race condition which allows to subvert the review.

**Comment by okurz** [comment]
> Sorry, but I don't think this is a good approach of merging things automatically, even with 2 approvals. And I'm afraid it may bring more troubles then benefits to our codebase. I've seen a lot of times when a PR received a couple of "Approvals" but then some new reviewer came and raised a valid point to change the PR and a lot of people change their mind.

True, that can happen. If someone feels like "I don't see anything wrong but I can not take the responsibility to approve" then just +1. We can also extend the configuration to ensure that a PR stays open for a minimum time or we only merge automatically if there are even more approvals than just 2 or we only keep the part where mergify would act on a specific label

> Also, nothing prevents from altering a PR after receiving "Approve". How to deal with such cases?

same as before. Reviewers either need to trust authors or only approve if everything is really fine. And if anyone wants to abuse the system I am sure we will teach them fast. I trust my colleagues … or at least I trust that I can find them when they mess up ;)

> One more thing is that it makes "Approve" button to be taken more serious then before.

And that's a good point, right? :)

**Comment by okurz** [comment]
> I'm really not a fan of enabling mergify for the tests, while it would be amazing and a lot of time saving, unless one can have a list of submitters whose's PR's can be automerged.
> 
> Reasoning for this: We run this on openSUSE and SLE as is, so if somebody misses checking a zipped file (of which we have many), or some random binary... we can end up in a bad place, as @Vogtinator mentions:
> 
> People can change things between approvals and add malicious code...
> 
> FUD aside, the other side of the problem is while we only would need one or two persons to approve, someone else's code might break. You break it, you fix it, is cool though.
> 
> So, from my pov:
> 
>     * Until we have an automated check that the pr is not adding compressed files
> 
>     * Until we have an automated check that there are no external things being loaded

That's a good point in general but you are applying too strict rules for a little helper for what reviewers so far also not always looked into.

How about only including the rule "automatic merge on special label" for a start?

File: .mergify.yml
Comment (mdoucha) [reply .mergify.yml:2]: > That's correct but IMHO has nothing to do with mergify. I think github can be configured to invalidate approvals in case any new commits change but that would defeat the purpose of allowing the author to do trivial, pre-agreed changes, like applying perltidy or fixing typos in comments or something. It's still in the responsibility of reviewers to only approve if the PR is _actually_ acceptable

Then you'd also have to forbid approving PRs which are marked as WIP. Without invalidating reviews on every push, the easiest way to sneak malicious code into the repo is:
- mark the PR as WIP
- wait for 2 reviews
- push malicious code
- remove WIP

You can have either automerge, or approvals that persist after code push. But not both.

**Comment by foursixnine** [comment]
> > I'm really not a fan of enabling mergify for the tests, while it would be amazing and a lot of time saving, unless one can have a list of submitters whose's PR's can be automerged.
> > Reasoning for this: We run this on openSUSE and SLE as is, so if somebody misses checking a zipped file (of which we have many), or some random binary... we can end up in a bad place, as @Vogtinator mentions:
> > People can change things between approvals and add malicious code...
> > FUD aside, the other side of the problem is while we only would need one or two persons to approve, someone else's code might break. You break it, you fix it, is cool though.
> > So, from my pov:
> > ```
> > * Until we have an automated check that the pr is not adding compressed files
> > 
> > * Until we have an automated check that there are no external things being loaded
> > ```
> 
> That's a good point in general but you are applying too strict rules for a little helper for what reviewers so far also not always looked into.
> 
> How about only including the rule "automatic merge on special label" for a start?

Let's have those two extra checks first, and then talk about this again, I'm not opening up to a free and big fat security hole, just because people believe in CI/CD a little bit too much.

**Comment by okurz** [comment]
> I'm really not a fan of enabling mergify for the tests, while it would be amazing and a lot of time saving, unless one can have a list of submitters whose's PR's can be automerged.

By the way, we can have that list of submitters, e.g. with

```
- author=@tests-maintainer
```

should I include that?

> Reasoning for this: We run this on openSUSE and SLE as is, so if somebody misses checking a zipped file (of which we have many), or some random binary... we can end up in a bad place

Within o3 we run with apparmor at least.

>     * Until we have an automated check that the pr is not adding compressed files

ok, that should be possible. So how about a check in github action that conducts this and if it fails then mergify would not merge automatically. Reviewers would be alerted because they see the red X and take extra care about the added file.

The check could be

```
git diff --name-only origin/master.. | xargs file --mime | grep 'application/.*\(zip\|compressed\)'
```

this detects not only plain "zip" files but gzip, bzip2, 7z, rar, etc.

>     * Until we have an automated check that there are no external things being loaded

Do you mean no external things loaded during CI test execution or within openQA test code? The first can be covered with executing tests with `unshare -r -n` which takes away network access and any reliance on external network would quickly be discovered. Or do you mean within the openQA test code? If that, how do you qualify "no external things being loaded" exactly?

**Comment by okurz** [comment]
Updated the PR:
* Split the changes into individual commits so if you like you can review the commits one by one and comment on each individually to better separate the discussions
* Now approvals will be automatically removed as soon as changes have been introduced so that approvals will not stick and hence no one can sneak in malicious code that way
* Changed the "automatic merge" based on all successful checks to instead just provide another status check instead of an automatic merge

File: .mergify.yml
Comment (okurz) [reply .mergify.yml]: ok. With https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455/commits/e9f9a6f18852d892735a55b935617c8ac671a37c now approvals will be removed if changes have been pushed. Also now there won't be an automatic merge just based on successful checks, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455/commits/2a7b923a361fed6fe6e6e81de28d46e728c5ddc2

File: .mergify.yml
Comment (pdostal) [reply .mergify.yml:9]: What about the `WIP` label? I found that [apps/wip](https://github.com/apps/wip) checks labels only in some kind of 'PRO' version..

**Comment by Vogtinator** [comment]
What I still don't get is what benefit mergify actually brings.

FWICT, instead of hitting the merge/auto-merge button it's enough to submit an approving review, and once the criteria are fulfilled, mergify will do the "work" of actually performing the merge. Why not just hit the merge button directly? Is it to make it harder to accidentially merge a PR which wasn't meant to be merged by accident? If so, this should not be done by mergify, but rather by extending the already existing PR checks mechanism.

The downsides of mergify are that there's more complexity involved, both in configuration and usage, and a third-party is trusted with full access to the repo.

File: .mergify.yml
Comment (okurz) [reply .mergify.yml]: ok, included "wip" label.

File: .mergify.yml
Comment (pevik) [reply .mergify.yml:32]: I'd prefer to write: Could you *please* fix it?, but if you think emoji is important, I don't have strong opinion. I'd just try to avoid utf8 symbols if possible (not long time ago copyright symbol was removed).

Comment (okurz) [reply .mergify.yml:32]: Sure, I can change it. I just don't see why we should treat os-autoinst-distri-opensuse any different from [os-autoinst](https://github.com/os-autoinst/os-autoinst/blob/master/.mergify.yml#L39), [openQA](https://github.com/os-autoinst/openQA/blob/master/.mergify.yml#L40), [scripts](https://github.com/os-autoinst/scripts/blob/master/.mergify.yml#L31), [openqa-review](https://github.com/os-autoinst/openqa_review/blob/master/.mergify.yml#L29), [os-autoinst-common](https://github.com/os-autoinst/os-autoinst-common/blob/master/.mergify.yml#L26)

File: .mergify.yml
Comment (okurz) [inline .mergify.yml]: "post_check:" is a premium plan feature. I changed that to "review:" so that mergify actually also approves

File: .mergify.yml
Comment (okurz) [inline .mergify.yml:27]: Changed that from "post_check" which is a premium plan feature to "review"

**Review by b10n1k — APPROVED** [review]
LGTM

**Comment by foursixnine** [comment]
@okurz mind if I close this PR?.

**Comment by okurz** [comment]
> @okurz mind if I close this PR?. 

Depends if you think if we should have this change included in general or not. I can change stuff if you have good suggestions

File: .mergify.yml
Comment (pdostal) [reply .mergify.yml:33]: I also vote for keeping that emoji as it makes the workflow a bit more colourful. and even screen readers nowadays recognise those.

**Comment by pdostal** [comment]
What if we disable the automerge feature and only let mergify to check and moderate?

**Comment by foursixnine** [comment]
> Depends if you think if we should have this change included in general or not. I can change stuff if you have good suggestions

In general I think we don't need this.

> What if we disable the automerge feature and only let mergify to check and moderate?

Thatś what the rest of the checks also do already :)

File: .mergify.yml
Comment (okurz) [reply .mergify.yml:33]: so let's keep it in this way, same as we have for other projects

**Comment by okurz** [comment]
> > Depends if you think if we should have this change included in general or not. I can change stuff if you have good suggestions
> 
> In general I think we don't need this.

Why not? It already helps us in multiple projects.

> > What if we disable the automerge feature and only let mergify to check and moderate?
> 
> Thatś what the rest of the checks also do already :)

Not quite. I could repeat the original description but I'd rather suggest you read it again and please reference that in your answer.

**Comment by okurz** [comment]
> I'm answering to your question above. On the why not, obviously nobody has cared about it or seen the need to have it for amost a year (even with the use of a tag)

I don't understand how you can apply that as reasoning. That would imply that basically any idea noted in a ticket or also any future pull request would be useless because "nobody did it so far hence it might is not useful enough"?!?

> If you have more questions, please refer to my [previous answer](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992518111)

you might have missed my suggestions and further questions in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992537335 trying to answer your ponits and my update in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992664089

**Comment by foursixnine** [comment]
> > I'm answering to your question above. On the why not, obviously nobody has cared about it or seen the need to have it for amost a year (even with the use of a tag)
> 
> I don't understand how you can apply that as reasoning. That would imply that basically any idea noted in a ticket or also any future pull request would be useless because "nobody did it so far hence it might is not useful enough"?!?

I suggest you to open a ticket then.

> 
> > If you have more questions, please refer to my [previous answer](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992518111)
> 
> you might have missed my suggestions and further questions in [#13455 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992537335) trying to answer your ponits and my update in [#13455 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992664089)

Didn't miss them, read them before replying... My standing is still the same.

**Review by pdostal — APPROVED** [review]
🚀

**Comment by kalikiana** [comment]
> @foursixnine I don't understand what you mean or what else you would want me to do. I guess unless other people chime in to help resolve our deadlock, maybe by voting, after sufficient waiting time I will close this PR unmerged.
> 
> @os-autoinst/tools-team what's your take on this?

- None of the 3 related tickets seems really related. The tickets are about adding automated verification runs and staging tests. Although I could see how you consider it best practice.
- The points from https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992537335 seem misplaced here since this PR is not implementing execution of openQA tests and afair we only validate the schedule so far. I would second those concerns otherwise.
- The title seems wrong (now). I don't see automatic merging in the config YAML

**Comment by foursixnine** [comment]
> @foursixnine I don't understand what you mean or what else you would want me to do. 

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#pullrequestreview-829957246

> I guess unless other people chime in to help resolve our deadlock

Sure

**Comment by okurz** [comment]
@kalikiana 
>     * The points from [Add mergify for additional checks and automatic merging based on special label #13455 (comment)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issuecomment-992537335) seem misplaced here since this PR is not implementing execution of openQA tests and afair we only validate the schedule so far. I would second those concerns otherwise.

This repo has unit tests and multiple static code checks so it's not just validation of the schedule file validity.

>     * The title seems wrong (now). I don't see automatic merging in the config YAML

That's correct. I removed that part for now.

> > @foursixnine I don't understand what you mean or what else you would want me to do.
> 
> See [#13455 (review)](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#pullrequestreview-829957246)

As stated in before:
1. In the current config there is no automatic merge anymore
2. I already answered your two suggestions about further automatic checks and wanted to have feedback on those before continuing with any implementation

**Comment by kalikiana** [comment]
> This repo has unit tests and multiple static code checks so it's not just validation of the schedule file validity.

Yes. And I'm saying I don't see the concerns with what we have right now.

**Comment by okurz** [comment]
@Vogtinator + @foursixnine can you please check the PR in the current state? The PR in its current state is adding additional checks and adding an "approve" based on if all rules are fulfilled so it's merely an additional information for now, not doing "auto-merge" on its own right now.

File: .mergify.yml
Comment (Vogtinator) [inline .mergify.yml:25]: i.e. mergify approves all requested reviews?

That effectively means only a single actual approval is needed.

**Comment by okurz** [comment]
> > @Vogtinator + @foursixnine can you please check the PR in the current state? The PR in its current state is adding additional checks and adding an "approve" based on if all rules are fulfilled so it's merely an additional information for now, not doing "auto-merge" on its own right now.
> 
> I see only a single benefit: If the PR is only waiting for checks to succeed, then this can be utilized as a "Merge after checks completed" mechanism. Otherwise, whoever triggers the last needed approval can just hit the button themselves.

I would appreciate if you can relate to what I have listed in the initial description
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13455#issue-1025017934
because I see more than just this single benefit.

> Mergify is a third-party for-profit company, we really should not give them full access to this repo or even start relying on them in workflows.

We already use mergify in multiple other projects in github.com/os-autoinst/ . I see no problem with a third-party company having full access to a fully free-software project. I agree that we shouldn't fully rely on them but as soon as they turn bad I see no problem to go to a different solution. The rules file is still a good way to codify rules. And GitHub is also a for-profit company and we trust them, right?

**Comment by Vogtinator** [comment]
> We already use mergify in multiple other projects in github.com/os-autoinst/ . I see no problem with a third-party company having full access to a fully free-software project.

full access -> read/write privileges, which can go completely unnoticed. Code added here runs on production systems as well as on developer machines. IMO there really needs to be more awareness of what those privileges actually mean.

>  I agree that we shouldn't fully rely on them but as soon as they turn bad I see no problem to go to a different solution. The rules file is still a good way to codify rules. And GitHub is also a for-profit company and we trust them, right?

I don't, but even if I did, the principle of least privilege still exists. It's better to only have to trust one than two entities.

**Comment by okurz** [comment]
Then there is no way this can be merged as I don't want to overrule you.

**Comment by Vogtinator** [comment]
> @Vogtinator By the way, if you have good suggestions how to keep equivalent functionality in os-autoinst repos but with a better approach I would appreciate to hear from you

The clean way is to enable "required checks", optionally adding more checks to cover what mergify would here.
With that option enabled, GH itself actually offers an auto-merge option. That unfortunately makes it impossible to merge something before all checks completed, so it would be ideal if the required checks don't take very long.

**Comment by okurz** [comment]
That's not giving what currently mergify provides though. And in particular the settings in github are not visible to non-maintainers

**Comment by foursixnine** [comment]
> 

That's the intention behind https://progress.opensuse.org/issues/124688 and once we have them, pr's tested directly on openQA

**Comment by foursixnine** [comment]
> > @Vogtinator By the way, if you have good suggestions how to keep equivalent functionality in os-autoinst repos but with a better approach I would appreciate to hear from you
> 
> The clean way is to enable "required checks", optionally adding more checks to cover what mergify would here. With that option enabled, GH itself actually offers an auto-merge option. That unfortunately makes it impossible to merge something before all checks completed, so it would be ideal if the required checks don't take very long.

While I think that the Mergify option might not be the solution we're looking for

- I am not a monster :) https://progress.opensuse.org/issues/138542

**Comment by okurz** [comment]
So you want to continue the discussion in https://progress.opensuse.org/issues/138542 and not here. I think that is just less efficient but ok. I answered in https://progress.opensuse.org/issues/138542#note-1

**Pull request closed without merging**
